### PR TITLE
[None][feat] BREAKING: add per-model transceiver runtime auto selection

### PR DESCRIPTION
**Pull request by nv-xtf**
## Description

Add per-model auto selection for the KV-cache transceiver runtime on the standard PyTorch model-loading path.

- Add `auto` as the default value of `CacheTransceiverConfig.transceiver_runtime`.
- Add `get_preferred_transceiver_runtime()` for models to declare their preferred runtime.
- Preserve explicit user settings (`CPP`, `PYTHON`, or `None`).
- Apply the Python runtime preference only when the effective backend is NIXL.
- Fall back to the C++ transceiver when the model does not provide a preference, the effective backend is incompatible, model resolution is unavailable, or the loading path does not use the standard PyTorch model loader.
- Preserve the original model preference when the resolved execution class is redirected, such as the MTP draft-model path.
- Prevent `get_model_defaults()` from implicitly creating or enabling `cache_transceiver_config`.
- Share DEFAULT backend resolution between runtime selection and transceiver creation.
- Update the telemetry manifest, generated documentation, and related tests.

No model opts into the Python transceiver in this PR, so the effective default remains the C++ transceiver until model-specific migrations are added.

## Model Opt-in

To provide an automatic preference for a model, override `get_preferred_transceiver_runtime()` in its `ForCausalLM` implementation:

```python
@classmethod
def get_preferred_transceiver_runtime(cls, pretrained_config=None):
    return "PYTHON"
```

Valid return values are `"PYTHON"`, `"CPP"`, or `None` (defer to the global default, i.e. the C++ transceiver). Shared implementation classes can inspect `pretrained_config.architectures` to return different preferences for different model architectures.

The user can always override the model preference explicitly:

```yaml
cache_transceiver_config:
  backend: NIXL
  transceiver_runtime: CPP  # CPP, PYTHON, or auto
```

Resolution outcomes:

| User setting      | Model preference | Effective backend | Result                             |
|-------------------|------------------|-------------------|------------------------------------|
| unset / `auto`    | `PYTHON`         | NIXL              | Python transceiver                 |
| unset / `auto`    | `PYTHON`         | UCX/MPI/MOONCAKE  | C++ transceiver (fallback, logged) |
| unset / `auto`    | `None`           | any               | C++ transceiver                    |
| `CPP` / `None`    | any              | any               | C++ transceiver (user wins)        |
| `PYTHON`          | any              | NIXL              | Python transceiver (user wins)     |
| `PYTHON`          | any              | non-NIXL          | Error (unchanged behavior)         |

`auto` only selects the runtime after the user enables `cache_transceiver_config`; it does not enable disaggregated serving by itself. On paths that skip the standard PyTorch model loader (e.g. AutoDeploy), `auto` falls back to the C++ transceiver — in mixed deployments, set `transceiver_runtime` explicitly on both sides to keep them aligned.

The preference must not be declared through `get_model_defaults()`, because model defaults must not create or enable a transceiver configuration (rejected at load time).

With `TLLM_LOG_LEVEL=INFO`, adoption is visible at startup: `Resolved transceiver_runtime='auto' to 'PYTHON' for GptOssForCausalLM.` followed by `Using KvCacheTransceiverV2`.

## Breaking Change

The default value of `CacheTransceiverConfig.transceiver_runtime` changes from `None` (C++ runtime) to `auto`. Existing explicit values (`CPP`, `PYTHON`, and `None`) remain supported. No model opts into the Python runtime in this PR, so the effective runtime still falls back to C++ unless a model-specific preference is added later.

## Test Coverage

```bash
LLM_MODELS_ROOT=/tmp python -m pytest -q tests/unittest/llmapi/test_llm_args.py::TestTransceiverRuntimeAutoResolution

python -m pytest -q tests/unittest/usage/test_llmapi_config_capture.py::test_collect_llm_api_config_captures_transceiver_runtime_categorical

LLM_MODELS_ROOT=<model-root> python -m pytest -q tests/unittest/llmapi/test_llm_args.py tests/unittest/usage/
```

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

## Summary by CodeRabbit

* **New Features**
  * Added an automatic option for transceiver runtime selection, letting the system choose the best runtime when not set explicitly.
  * Expanded support for an additional runtime value in configuration capture and validation.

* **Bug Fixes**
  * Improved configuration loading so automatic runtime settings are resolved consistently across model setups and checkpoint-based workflows.
  * Prevented model defaults from overriding transceiver configuration in unsupported ways.

* **Documentation**
  * Updated telemetry/configuration docs to reflect the new runtime option and updated field counts.

### CODE REVIEW

**Review by Shixiaowei02 — APPROVED** [review]

**Review by QiJune — APPROVED** [review]
LGTM

File: tensorrt_llm/_torch/models/modeling_utils.py [hunk 1]
```diff
@@ -605,9 +609,41 @@ def get_model_defaults(cls, llm_args: 'TorchLlmArgs') -> dict:
...
+
+        Note: ``cache_transceiver_config`` is rejected here (enforced at
+        load time) — the deep-merge could materialize or silently enable a
+        transceiver config the user did not turn on. Use
+        :meth:`get_preferred_transceiver_runtime` instead.
         """
         return {}
 
+    @classmethod
+    def get_preferred_transceiver_runtime(
```
Comment (litaotju) [inline tensorrt_llm/_torch/models/modeling_utils.py:621]: I dont fully understand this design. 

I think our migration status where we want to be model specific is just a temp status. 

Finally we want to switch all the models to Python right?

Do we need to let per model to override this one or it should be a cental place maintained by dis-agg people who are working a per model migration?

**Review by litaotju — CHANGES_REQUESTED** [review]

File: tensorrt_llm/_torch/models/modeling_utils.py (hunk 1 above)
Comment (nv-xtf) [reply tensorrt_llm/_torch/models/modeling_utils.py:621]: Thanks for raising this. I agree that per-model selection is a transitional mechanism, and the long-term goal is to use Python for all supported model/backend combinations. During migration, I think each model owner should set the preference after validating correctness and performance, while the disaggregated-serving team owns the common resolver and migration criteria. Once Python is safe as the global default, we can remove the per-model preferences and resolve auto to Python globally.

Comment (litaotju) [reply tensorrt_llm/_torch/models/modeling_utils.py:621]: "each model owner should set the preference"? I think our plan is that dis-agg team working on verifying the set of core models like DS/Kimi/GPT-OSS etc. 

And after that we default all to V2, and on-demand fix things if V2 has reported perf/functional issue. The model developer shall not change any of the preference. 

Once we switch to V2, all new models shall ONLY use V2.

cc @Shixiaowei02 for this discussion. We can also talk offline about this one.

**Review by tburt-nv — APPROVED** [review]

**Review by litaotju — APPROVED** [review]
